shark-graph/src/main/java/shark/HeapValue.kt

```diff
@@ -73,12 +73,6 @@ class HeapValue(
   val asLong: Long?
     get() = if (holder is LongHolder) holder.value else null
 
-  /**
-   * This [HeapValue] as a [Long] if it represents an object reference, or null otherwise.
-   */
-  val asObjectId: Long?
-    get() = if (holder is ReferenceHolder) holder.value else null
```

Comment:
Ah, didn't notice it is public API. Again, haven't done very thorough measuring, but after this and other change the % of `Long.valueOf()` calls dropped.
I can try to count how many times this method is used and how big of an impact it is.
We can keep it but replace internal usages with equivalents that don't use this method to prevent boxing